### PR TITLE
Rename package: [OpenTelemetry.Contrib.Extensions.AWSXRay] to [OpenTelemetry.Extensions.AWS]

### PR DESCRIPTION
**Pull request by rypdal**
Name of the package OpenTelemetry.Contrib.Extensions.AWSXRay is not consistent with current naming conventions. 
1. Contrib suffix should be removed
2. AWSXRay should be replaced by AWS (see discussion https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/875#issuecomment-1452338635)

Package has been renamed to OpenTelemetry.Extensions.AWS. Namespaces, etc have been aligned accordingly.

This PR is result of discussions in the other PR: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1224#issuecomment-1581162972

EDIT: Fixes: #1066

### CODE REVIEW

**Comment by rypdal** [comment]
@srprash, @atshaw43, @Kielek and @Oberon00: I would be very thankful for your review and comments. 
This PR is the pre-requisite for releasing of the new OpenTelemetry.Extensions.AWS package, which is used by OpenTelemetry.Contrib.Instrumentation.AWS and OpenTelemetry.Instrumentation.AWSLambda packages.

**Review by Kielek — APPROVED** [review]
I have pushed some minor changes directly to your branch.

LGTM.

File: .github/ISSUE_TEMPLATE/comp_contrib_extensions_aws.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
 ---
-name: OpenTelemetry.Contrib.Extensions.AWSXRay
-about: Issue with OpenTelemetry.Contrib.Extensions.AWSXRay
-labels: comp:contrib.extensions.awsxray
+name: OpenTelemetry.Extensions.AWS
+about: Issue with OpenTelemetry.Extensions.AWS
+labels: comp:extensions.aws
```
Comment (Oberon00) [inline .github/ISSUE_TEMPLATE/comp_contrib_extensions_aws.md:4]: If this is merge, don't forget to rename the label on GitHub as well.

**Review by srprash — APPROVED** [review]

**Review by Oberon00 — CHANGES_REQUESTED** [review]
Make sure not to merge, there is a mistake here: The goal was to remove the "Contrib" part from the name, but this renames AWSXray to AWS which would then ultimately lead to conflicts.

**Comment by Oberon00** [comment]
Or maybe it's not a mistake, I'm a bit confused now. I'm pretty sure we want to remove "contrib" but why do we want to remove "xray"? Will this then not be easy to confuse with the other "AWS" package (Extensions vs Instrumentation)?

**Comment by rypdal** [comment]
@Oberon00:  I think renaming to OpenTelemetry.Extensions.AWS should be fine as it will no conflict with OpenTelemetry.Instrumentation.AWS. Also it more or less reflects what is this package for.

**Pull request merged**